src/main/scala/io/iohk/ethereum/jsonrpc/PersonalService.scala

```diff
@@ -87,6 +91,19 @@ class PersonalService(
     Right(LockAccountResponse(true))
   }
 
+  def ecRecover(req: EcRecoverRequest): ServiceResponse[EcRecoverResponse] = Future {
+    import req._
+    val prefixed: Array[Byte] =
+      Array[Byte](19.toByte) ++
+        s"Ethereum Signed Message:\n${message.length}".map(c => c.toByte).toArray[Byte] ++
+        message.toArray[Byte]
```

Comment:
[In geth](https://github.com/ethereum/go-ethereum/blob/1a0eb903f1e84772b14f399e519b4a3408137fdc/internal/ethapi/api.go#L361) the character is in hexadecimal, here you have a decimal. Could that be the problem?